wagtailio/core/blocks.py

```diff
@@ -64,13 +66,31 @@ class Meta:
         template = "core/blocks/page_or_external_link_block.html"
 
 
+class BannerVideoBlock(media_blocks.AbstractMediaChooserBlock):
+    pass
+
+
 class BannerBlock(blocks.StructBlock):
     title = blocks.CharBlock(max_length=128)
     sub_title = blocks.CharBlock(max_length=128)
-    image = ImageChooserBlock()
+    image = ImageChooserBlock(required=False)
+    video = BannerVideoBlock(required=False)
     background = ImageChooserBlock(required=False)
     links = blocks.ListBlock(PageOrExternalLinkBlock())
 
+    def clean(self, value):
+        errors = {}
+
+        if not value.get('image') and not value.get('video'):
+            error_message = 'You must select either an image or a video.'
+            errors['image'] = ErrorList([error_message])
+            errors['video'] = ErrorList([error_message])
+
+        if errors:
+            raise ValidationError('Validation error in StructBlock', params=errors)
```

Comment:
question: does "Validation error in StructBlock" get shown to the user?
if so, a more suggestive message would be more helpful. If not, do ignore me :)

Reply:
This doesn't get displayed anywhere, in fact, Wagtail uses this as the error message too: https://github.com/wagtail/wagtail/blob/main/wagtail/core/blocks/struct_block.py#L20

Note that ``StructBlockValidationError`` was introduced recenly and we can't use that on wagtail.io yet

Reply:
thank you @kaedroho, should not have been lazy!